src/util/FioAddressUtils.ts

```diff
@@ -356,10 +360,28 @@ const updatePublicAddresses = async (
   try {
     edgeTx = await fioMakeSpend(fioWallet, action, {
       fioAddress,
-      publicAddresses
+      publicAddresses: publicAddresses.map(p => {
+        const fioAssets = infoServerData.rollup?.fioAssets ?? FIO_ASSET_MAP
+        const pluginId = Object.keys(fioAssets).find(pid => fioAssets[pid].chainCode === p.chain_code)
```

Comment:
I'm noticing this lookup pattern where our constraint is not only that the key in the `fioAssets` map is a match, but also the `chainCode` field matches. Why is that?

Also, could we please have `pid` change to `pluginId` since `pid` is easily confused with `process ID`? This is an optional request. I get tripped up on acronym usage unless they're well established.

Reply:
We're going from fio "chain code" to Edge pluginId